paymentsheet/src/main/java/com/stripe/android/paymentsheet/DefaultPaymentSheetLauncher.kt

```diff
@@ -12,7 +19,17 @@ import org.jetbrains.annotations.TestOnly
  */
 internal class DefaultPaymentSheetLauncher(
     private val activityResultLauncher: ActivityResultLauncher<PaymentSheetContract.Args>,
-) : PaymentSheetLauncher {
+    application: Application
+) : PaymentSheetLauncher, Injector {
+    private val paymentSheetLauncherComponent: PaymentSheetLauncherComponent =
+        DaggerPaymentSheetLauncherComponent.builder().application(application).build()
+
+    @InjectorKey
+    private val injectorKey: Int = WeakMapInjectorRegistry.nextKey()
+
+    init {
```

Comment:
It feels like there is a bunch of duplicate code surrounding the WeakMapInjector, I wonder if there is more we can do to have an interface or base class that helps enforce some behavior when you need this class.